homeassistant/components/nsw_fuel_station/sensor.py

```diff
@@ -45,134 +49,71 @@
-            except FuelCheckError as exc:
-                self.error = str(exc)
-                _LOGGER.error(
-                    "Failed to fetch NSW Fuel station reference data. %s", exc
-                )
-                return
-
-        try:
-            self._data = self._client.get_fuel_prices_for_station(self.station_id)
-        except FuelCheckError as exc:
-            self.error = str(exc)
-            _LOGGER.error("Failed to fetch NSW Fuel station price data. %s", exc)
-
-    def for_fuel_type(self, fuel_type: str):
-        """Return the price of the given fuel type."""
-        if self._data is None:
-            return None
-        return next(
-            (price for price in self._data if price.fuel_type == fuel_type), None
-        )
-
-    def get_available_fuel_types(self):
-        """Return the available fuel types for the station."""
-        return [price.fuel_type for price in self._data]
-
-    def get_station_name(self) -> str:
-        """Return the name of the station."""
-        if self._station_name is None:
-            name = None
-            if self._reference_data is not None:
-                name = next(
-                    (
-                        station.name
-                        for station in self._reference_data.stations
-                        if station.code == self.station_id
-                    ),
-                    None,
-                )
-
-            self._station_name = name or f"station {self.station_id}"
-
-        return self._station_name
-
-
-class StationPriceSensor(SensorEntity):
+class StationPriceSensor(CoordinatorEntity, SensorEntity):
     """Implementation of a sensor that reports the fuel price for a station."""
 
-    def __init__(self, station_data: StationPriceData, fuel_type: str) -> None:
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[StationPriceData],
+        station_id: int,
+        fuel_type: str,
+    ) -> None:
         """Initialize the sensor."""
-        self._station_data = station_data
+        super().__init__(coordinator)
+
+        self._station_id = station_id
         self._fuel_type = fuel_type
 
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._station_data.get_station_name()} {self._fuel_type}"
+        station_name = self._get_station_name()
+        return f"{station_name} {self._fuel_type}"
 
     @property
     def state(self) -> float | None:
         """Return the state of the sensor."""
-        price_info = self._station_data.for_fuel_type(self._fuel_type)
-        if price_info:
-            return price_info.price
+        if self.coordinator.data is None:
+            return None
 
-        return None
+        prices = self.coordinator.data.prices
+        return prices.get((self._station_id, self._fuel_type))
+
+    @property
+    def available(self) -> bool:
```

Comment:
CoordinatorEntity already provides this.  This can be removed

Reply:
Fixed, nice catch!